examples/wakaama/dtlsconnection.c

```diff
@@ -0,0 +1,570 @@
+   if (cnx != NULL) {
+      lwm2m_handle_packet(cnx->lwm2mH, (uint8_t *)data, len, (void *)cnx);
+      return 0;
+   }
+   return -1;
+}
+/**************************   TinyDTLS Callbacks Ends ************************/
+/**************************   RIOT GNRC  *************************************/
+
+/**
+ * @brief This will try to transmit using only GNRC stack (non-socket).
+ */
+static int gnrc_sending(char *addr_str, char *data, size_t data_len,
+                        unsigned short rem_port) {
+   int iface;
+   ipv6_addr_t addr;
+   gnrc_pktsnip_t *payload, *udp, *ip;
+
+   /* get interface, if available */
+   iface = ipv6_addr_split_iface(addr_str);
+   if ((iface < 0) && (gnrc_netif_numof() == 1)) {
+      iface = gnrc_netif_iter(NULL)->pid;
+   }
+   /* parse destination address */
+   if (ipv6_addr_from_str(&addr, addr_str) == NULL) {
+      puts("Error: unable to parse destination address");
+      return -1;
+   }
+
+   payload = gnrc_pktbuf_add(NULL, data, data_len, GNRC_NETTYPE_UNDEF);
+
+   if (payload == NULL) {
+      puts("Error: unable to copy data to packet buffer");
+      return -1;
+   }
+
+   /* allocate UDP header */
+   udp = gnrc_udp_hdr_build(payload, DEFAULT_PORT, rem_port);
+   if (udp == NULL) {
+      puts("Error: unable to allocate UDP header");
+      gnrc_pktbuf_release(payload);
+      return -1;
+   }
+   /* allocate IPv6 header */
+   ip = gnrc_ipv6_hdr_build(udp, NULL, &addr);
+   if (ip == NULL) {
+      puts("Error: unable to allocate IPv6 header");
+      gnrc_pktbuf_release(udp);
+      return -1;
+   }
+   /* add netif header, if interface was given */
+   if (iface > 0) {
+      gnrc_pktsnip_t *netif = gnrc_netif_hdr_build(NULL, 0, NULL, 0);
+
+      ((gnrc_netif_hdr_t *)netif->data)->if_pid = (kernel_pid_t)iface;
+      LL_PREPEND(ip, netif);
+   }
+   /* send packet */
+
+   DEBUG("Sending stuff via gnrc_sending(%s, %d)\n", addr_str, rem_port);
+
+   /*
+    * WARNING: Too fast and the nodes dies in middle of retransmissions.
+    *         This issue appears in the FIT-Lab (m3 motes).
+    */
+   xtimer_usleep(500000);
+
+   /* Probably this part will be removed.  **/
+   if (!gnrc_netapi_dispatch_send(GNRC_NETTYPE_UDP, GNRC_NETREG_DEMUX_CTX_ALL,
+                                  ip)) {
+      puts("Error: unable to locate UDP thread");
+      gnrc_pktbuf_release(ip);
+      return -1;
+   }
+
+   return 1;
+}
+
+/**************************   RIOT GNRC Ends  ********************************/
+
+static dtls_handler_t cb = {
+    .write = send_to_peer,
+    .read = read_from_peer,
+    .event = NULL,
+    //#ifdef DTLS_PSK
+    // FIXME: other key stuff
+    .get_psk_info = get_psk_info,
+    //#endif /* DTLS_PSK */
+    //#ifdef DTLS_ECC
+    //  .get_ecdsa_key = get_ecdsa_key,
+    //  .verify_ecdsa_key = verify_ecdsa_key
+    //#endif /* DTLS_ECC */
+};
+
+dtls_context_t *get_dtls_context(dtls_connection_t *connList) {
+   if (dtlsContext == NULL) {
+      dtls_init();
```

Comment:
This should be moved to main.c 
dtls_init() must be called only once time by reboot (This is not explicit  yet in  `master:examples/dtls-echo`). 

Otherwise, there is risk of unexpected behavior when a second or posterior context is created